src/components/PostDetails.tsx

```diff
@@ -1,106 +1,132 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
+import { Post } from '../types/Post';
 import { NewCommentForm } from './NewCommentForm';
+import { getComments, addComment, deleteComment } from '../utils/fetchClient';
+import { Comment, CommentData } from '../types/Comment';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
-
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+type Props = {
+  currPost: Post | null;
+  openModal: boolean;
+  setOpenModal: (value: boolean) => void;
+};
 
-        <div className="block">
-          <Loader />
+export const PostDetails: React.FC<Props> = ({
+  currPost,
+  openModal,
+  setOpenModal = () => {},
```

Comment:
Setting a default value for `setOpenModal` can lead to unexpected behavior if the parent component does not pass this prop explicitly. Consider removing the default value to ensure that the parent component always provides this function.